zerver/lib/types.py

```diff
@@ -221,6 +221,49 @@ class NeverSubscribedStreamDict(TypedDict):
     subscribers: NotRequired[List[int]]
 
 
+class StreamClientDict(TypedDict):
+    """Stream information provided to Zulip clients as a dictionary via API.
+    It should contain all the fields specified in `zerver.models.Stream.API_FIELDS`
+    with few exceptions and possible additional fields.
+    """
+
+    date_created: int
+    description: str
+    first_message_id: Optional[int]
+    history_public_to_subscribers: bool
+    stream_id: int  # `stream_id`` represents `id` of the `Stream` object in `API_FIELDS`
+    invite_only: bool
+    is_web_public: bool
+    message_retention_days: Optional[int]
+    name: str
+    rendered_description: str
+    stream_post_policy: int
```

Comment:
How are these fields ordered? I'd generally prefer either alphabetical or something thematic (E.g. `rendered_description` next to `description`).

Reply:
I believe that it's ordered the same way as in `Stream.API_FIELDS`. Can do a prep commit sorting them both.

Reply:
`API_FIELDS` was ordered alphabetically. I updated the dict fields to make it order that way as well.